mono/utils/mach-support-x86.c

```diff
@@ -151,64 +135,9 @@ mono_mach_arch_set_thread_states (thread_port_t thread, thread_state_t state, ma
 #endif	
 }
 
-void *
-mono_mach_get_tls_address_from_thread (pthread_t thread, pthread_key_t key)
-{
-	/* OSX stores TLS values in a hidden array inside the pthread_t structure
-	 * They are keyed off a giant array from a known offset into the pointer.  This value
-	 * is baked into their pthread_getspecific implementation
-	 */
-	intptr_t *p = (intptr_t *) thread;
-	intptr_t **tsd = (intptr_t **) ((char*)p + tls_vector_offset);
-	g_assert (tls_vector_offset != -1);
-
-	return (void *) &tsd [key];	
-}
-
-void *
-mono_mach_arch_get_tls_value_from_thread (pthread_t thread, guint32 key)
-{
-	return *(void**)mono_mach_get_tls_address_from_thread (thread, key);
-}
-
 void
 mono_mach_init (pthread_key_t key)
```

Comment:
Why not removing fully the function?

Reply:
Well, I can... it just seemed like this one could have some use in the future.